src/parser/mod.rs

```diff
@@ -896,35 +887,6 @@ impl<'a> Parser<'a> {
         Ok(expr)
     }
 
-    pub fn parse_interval_expr(&mut self) -> Result<Expr, ParserError> {
```

Comment:
These are "moved" to after `parse_interval`, so the ordering makes more sense, but really they're completely rewritten.